sn_cli/src/subcommands/files.rs

```diff
@@ -137,11 +137,49 @@ pub(crate) async fn files_cmds(
                 );
             }
 
-            let download_dir = dirs_next::download_dir().unwrap_or(root_dir.to_path_buf());
+            let mut download_dir = root_dir.to_path_buf();
+            let mut download_file_name = file_name.clone();
+            if file_name.is_some() {
+                // file_name may direct the downloaded data to:
+                //
+                // the current directory (just a filename)
+                // eg safe files download myfile.txt ADDRESS
+                //
+                // a directory relative to the current directory (relative filename)
+                // eg safe files download my/relative/path/myfile.txt ADDRESS
+                //
+                // a directory relative to root of the filesystem (absolute filename)
+                // eg safe files download /home/me/mydir/myfile.txt ADDRESS
+                let file_name_val = file_name.expect("file_name unspecified");
+                let file_name_path = Path::new(&file_name_val);
+                if file_name_path.is_dir() {
+                    return Err(eyre!("Cannot download file to path: {:?}", file_name_val));
+                }
+                let file_name_dir = file_name_path.parent();
+                if file_name_dir.is_none() {
+                    // just a filename, use the current_dir
+                    download_dir = std::env::current_dir().unwrap_or(root_dir.to_path_buf());
+                } else if file_name_path.is_relative() {
+                    // relative to the current directory. Make the relative path
+                    // into an absolute path by joining it to current_dir
+                    let relative_dir =
+                        file_name_dir.expect("file_name relative directory unspecified");
```

Comment:
Same here, this expect could be removed with an `if let Some(file_name_dir) = file_name_dir` clause